meos/src/point/pgis_types.c

```diff
@@ -1721,30 +1720,41 @@ geom_in(char *str, int32 typmod)
   lwgeom_parser_result_init(&lwg_parser_result);
 
   /* Empty string. */
-  if ( str1[0] == '\0' ) {
+  if (str[0] == '\0')
+  {
     meos_error(ERROR, MEOS_ERR_TEXT_INPUT, "parse error - invalid geometry");
     return NULL;
   }
 
   /* Starts with "SRID=" */
+  const char *str1 = str;
   if (pg_strncasecmp(str1, "SRID=", 5) == 0)
   {
     /* Roll forward to semi-colon */
-    char *tmp = str1;
-    while (tmp && *tmp != ';')
-      tmp++;
+    int delim = 0;
+    while ((str1)[delim] != ';' && (str1)[delim] != '\0')
+      delim++;
+    if ((str1)[delim] == '\0')
+    {
+      meos_error(ERROR, MEOS_ERR_TEXT_INPUT,
+        "Could not parse geometry value: %s", str);
+      return false;
+    }
 
-    /* Check next character to see if we have WKB  */
-    if (tmp && *(tmp+1) == '0')
+    /* Check next character to see if we have WKB */
+    if ((str1)[delim + 1] == '0')
     {
+      char *tmp = palloc(sizeof(char) * (delim + 1));
+      strncpy(tmp, str1, delim);
       /* Null terminate the SRID= string */
-      *tmp = '\0';
+      tmp[delim] = '\0';
       /* Set str1 to the start of the real WKB */
-      str1 = tmp + 1;
-      /* Move tmp to the start of the numeric part */
-      tmp = str + 5;
+      str1 += delim + 1;
+      /* Move str to the start of the numeric part */
+      tmp += 5;
       /* Parse out the SRID number */
       srid = atoi(tmp);
+      pfree(tmp);
```

Comment:
Apparently calling `pfree(tmp)` after having done `tmp += 5` produces the following warning:
```
/home/maxime/Documents/phd/code/MobilityDB/meos/src/point/pgis_types.c: In function ‘geom_in’:
/home/maxime/Documents/phd/code/MobilityDB/meos/postgres/postgres.h:75:15: warning: ‘free’ called on pointer ‘tmp’ with nonzero offset 5 [-Wfree-nonheap-object]
   75 | #define pfree free
/home/maxime/Documents/phd/code/MobilityDB/meos/src/point/pgis_types.c:1757:7: note: in expansion of macro ‘pfree’
 1757 |       pfree(tmp);
      |       ^~~~~
/home/maxime/Documents/phd/code/MobilityDB/meos/postgres/postgres.h:73:16: note: returned from ‘malloc’
   73 | #define palloc malloc
/home/maxime/Documents/phd/code/MobilityDB/meos/src/point/pgis_types.c:1747:19: note: in expansion of macro ‘palloc’
 1747 |       char *tmp = palloc(sizeof(char) * (delim + 1));
```
Not sure what the actual behaviour is, but this should probably be fixed in a future PR.